features/cellular/framework/targets/UBLOX/AT/UBLOX_AT_CellularNetwork.cpp

```diff
@@ -0,0 +1,309 @@
+/*
+ * Copyright (c) 2017, Arm Limited and affiliates.
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "UBLOX_AT_CellularNetwork.h"
+#include "UBLOX_AT_CellularStack.h"
+
+using namespace mbed;
+
+UBLOX_AT_CellularNetwork::UBLOX_AT_CellularNetwork(ATHandler &atHandler) : AT_CellularNetwork(atHandler)
+{
+    _op_act = RAT_UTRAN;
+    // The authentication to use
+    _auth = NSAPI_SECURITY_UNKNOWN;
+}
+
+UBLOX_AT_CellularNetwork::~UBLOX_AT_CellularNetwork()
+{
+    disconnect_modem_stack();
+
+    if (_connection_status_cb) {
+        _connection_status_cb(NSAPI_EVENT_CONNECTION_STATUS_CHANGE, NSAPI_ERROR_CONNECTION_LOST);
+    }
+}
+
+NetworkStack *UBLOX_AT_CellularNetwork::get_stack()
+{
+    if (!_stack) {
+        _stack = new UBLOX_AT_CellularStack(_at, _cid, _ip_stack_type);
+    }
+    return _stack;
+}
+
+bool UBLOX_AT_CellularNetwork::get_modem_stack_type(nsapi_ip_stack_t requested_stack)
+{
+    return requested_stack == IPV4_STACK ? true : false;
+}
+
+bool UBLOX_AT_CellularNetwork::has_registration(RegistrationType reg_type)
+{
+    return (reg_type == C_REG || reg_type == C_GREG);
+}
+
+nsapi_error_t UBLOX_AT_CellularNetwork::set_access_technology_impl(RadioAccessTechnology opRat)
```

Comment:
Then can we do like that 

    switch(opRat) {
#if defined(TARGET_UBLOX_C030_U201) || defined(TARGET_UBLOX_C030_U201)
		case RAT_UTRAN:
			break;
		case RAT_EGPRS:
			break;
		case RAT_E_UTRAN:
			break;
#elif defined(TARGET_UBLOX_C030_R410M)
		case RAT_CATM1:
			break;
#elif defined(TARGET_UBLOX_C030_R410M) || defined(TARGET_UBLOX_C030_N211)
		case RAT_NB1:
			break;
#endif
		default: {
			_op_act = RAT_UNKNOWN;
			return NSAPI_ERROR_UNSUPPORTED;
		}
    }

Reply:
Yes, I think something like that would be good.

Reply:
Just thinking so what about C027? 
(Please ignore this comment if not relevant)

Reply:
Good point, this does need to work on C027.

Reply:
I guess the second condition here was meant to be TARGET_UBLOX_C027 (more of a typo) : `#if defined(TARGET_UBLOX_C030_U201) || defined(TARGET_UBLOX_C030_U201)` . 

Point to note, SARA-G350 is 2G only so UTRAN, E-UTRAN should return error so just requires further classification.

Reply:
Yes, its typo mistake.

Reply:
This should also check all values in the enum RadioAccessTechnology : https://github.com/ARMmbed/mbed-os/blob/master/features/cellular/framework/API/CellularNetwork.h#L119. 
Otherwise, for example, RAT_HSDPA/RAT_HSUPA would return error although they are valid for SARA-U201.